src/marqo/s2_inference/clip_utils.py

```diff
@@ -192,7 +192,7 @@ class CLIP:
     conveniance class wrapper to make clip work easily for both text and image encoding
     """
 
-    def __init__(self, model_type: str = "ViT-B/32", device: str = 'cpu',  embedding_dim: int = None,
+    def __init__(self, model_type: str = "ViT-B/32", device: str = None,  embedding_dim: int = None,
                             truncate: bool = True, **kwargs) -> None:
```

Comment:
Optional[str] = None. 

something from chatgpt

Yes, using `device: Optional[str] = None` is generally better than `device: str = None` when you want to indicate that the `device` parameter is optional and can accept `None` as a valid value.

By using `Optional[str]`, you explicitly annotate that the parameter can either be a string or `None`. This conveys the intention that the parameter is optional and provides better clarity to other developers reading the code or using the function.

Using `Optional` from the `typing` module is a more precise and expressive way to indicate optional parameters in Python. It enhances readability and helps convey the intended contract of the function.

Here's an example demonstrating the usage of `Optional[str]`:

```python
from typing import Optional

def some_function(device: Optional[str] = None):
    # Perform operations based on the device
    if device is None:
        print("No device specified")
    else:
        print(f"Using device: {device}")

# Example usage
some_function()  # No device specified
some_function(device="cuda:0")  # Using device: cuda:0
some_function(device=None)  # No device specified
```

In the example above, the `device` parameter is annotated as `Optional[str]`, indicating that it can accept a string or `None`. Inside the function, the presence of `None` can be checked using an `if` statement to handle the case where no device is specified.

Using `Optional[str]` provides better clarity and avoids potential confusion about the intention of the parameter compared to simply using `str`.

Reply:
I do not want to indicate that device is optional, because it really shouldn't be. I only left a default `None` because otherwise it would require rearranging the parameters, thus making big downstream changes. Maybe this can be done in a separate PR. cc @pandu-k if this is alright.

Reply:
Ok, I can add that condition for all these functions to manually raise an error when device is None!